### PR TITLE
Azure Logs analytics

### PR DESCRIPTION
**Pull request by Tpo76**
I didn't find a way to overload the header "content-type" between authentication and the query request in the api.pm file correctly.
So this one is more a workaround to make this check works. If you know how to do it, I'll make the change.

Possible enhancement: matching string for threshold.

### CODE REVIEW

**Comment by Sims24** [comment]
Hi, 

Thanks, could you provide by mail curl commands from authentication to the query toward .io log endpoint please ? 

Think it could be great to check what custom query we may offer through the cli params.

File: cloud/azure/custom/api.pm [hunk 1]
```diff
@@ -163,7 +163,7 @@ sub settings {
 sub get_access_token {
     my ($self, %options) = @_;
 
-    my $has_cache_file = $options{statefile}->read(statefile => 'azure_api_' . md5_hex($self->{subscription}) . '_' . md5_hex($self->{tenant}) . '_' . md5_hex($self->{client_id}));
+    my $has_cache_file = $options{statefile}->read(statefile => 'azure_api_' . md5_hex($self->{subscription}) . '_' . md5_hex($self->{tenant}) . '_' . md5_hex($self->{client_id}) . '_' . md5_hex($self->{management_endpoint}));
```
Comment (garnier-quentin) [inline cloud/azure/custom/api.pm:166]: Why do you add $self->{management_endpoint} ?

**Comment by garnier-quentin** [comment]
I have fixed the issue with the content-type. Like Sims says, we need to the debug of your endpoint log please.

**Comment by garnier-quentin** [comment]
I have done the change. Could you test with that version please ?

**Pull request closed without merging**